src/oci_util.erl

```diff
@@ -165,25 +165,26 @@ to_num(Num) ->
     {S,Ex,Dgt} = case string:tokens(Num, ".") of
                   % Negative
                   [[$-|W],F] ->
-                      {E,D} = ed(W,F),
+                      {E,D} = ed(wf, {W,F}),
                       {$-,E,D};
                   [[$-|W]] ->
-                      {E,D} = ed(W,""),
+                      {E,D} = ed(w, W),
                       {$-,E,D};
                   % Positive
                   [W,F] ->
-                      {E,D} = ed(W,F),
+                      {E,D} = ed(wf, {W,F}),
                       {$+,E,D};
                   [W] ->
-                      {E,D} = ed(W,""),
+                      {E,D} = ed(w, W),
                       {$+,E,D}
               end,
-    L = length(D)+1,
+    L = length(Dgt)+1,
     case S of
         $- ->
-            Digits = list_to_binary([Di+101||Di<-Dgt]),
+            Digits = list_to_binary([-Di+101||Di<-Dgt]),
             Exp = bnot(Ex+128+65),
-            << L, 0:1, Exp:7, Digits/bytes
+            L2 = if L < 21 -> L + 1; true -> L end,
+            << L2, 0:1, Exp:7, Digits/bytes
                , if L < 21 -> <<102>>; true-> <<>> end/bytes >>;
```

Comment:
L2 can be inlined (since its not used elsewhere:
```erlang
<< if L < 21 -> L + 1; true -> L end,
   0:1, Exp:7, Digits/bytes,
   if L < 21 -> <<102>>; true-> <<>> end/bytes >>;
```